src/main/java/org/example/bookshop/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.example.bookshop.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.bookshop.dto.category.CategoryDto;
+import org.example.bookshop.dto.category.CreateCategoryRequestDto;
+import org.example.bookshop.dto.category.UpdateCategoryRequestDto;
+import org.example.bookshop.mapper.CategoryMapper;
+import org.example.bookshop.model.Category;
+import org.example.bookshop.repository.CategoryRepository;
+import org.example.bookshop.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable)
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryMapper.toDto(categoryRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id:" + id)));
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        return categoryMapper.toDto(
+                categoryRepository.save(
+                        categoryMapper.toEntity(requestDto)));
```

Comment:
Let's improve readability
```suggestion
        Category category = categoryMapper.toEntity(requestDto);
        return categoryMapper.toDto(categoryRepository.save(category));
```